internal/copy/copy_dir.go

```diff
@@ -87,22 +85,35 @@ func CopyDir(dst, src string) error {
 			return nil
 		}
 
-		// If the current path is a symlink, recreate the symlink relative to
-		// the dst directory
-		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
-			target, err := os.Readlink(path)
-			if err != nil {
-				return fmt.Errorf("failed to read symlink %q: %w", path, err)
+		errg.Go(func() error {
+
+			// we don't want to try and copy the same file over itself.
+			if eq, err := SameFile(path, dstPath); err != nil {
+				return fmt.Errorf("failed to check if files are the same: %w", err)
+			} else if eq {
+				return nil
 			}
 
-			if err := os.Symlink(target, dstPath); err != nil {
-				return fmt.Errorf("failed to create symlink %q: %w", dstPath, err)
+			// If the current path is a symlink, recreate the symlink relative to
+			// the dst directory
+			if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+				target, err := os.Readlink(path)
+				if err != nil {
+					return fmt.Errorf("failed to read symlink %q: %w", path, err)
+				}
+
+				if err := os.Symlink(target, dstPath); err != nil {
+					return fmt.Errorf("failed to create symlink %q: %w", dstPath, err)
+				}
+				return nil
 			}
-			return nil
-		}
-		return copyFile(dstPath, path, info.Mode())
+			return copyFile(dstPath, path, info.Mode())
+		})
+		return nil
 	}
-	return filepath.Walk(src, walkFn)
+	err = filepath.Walk(src, walkFn)
```

Comment:
Curious as to why you're using `filepath.Walk` over `filepath.WalkDir` here. Would you consider adjusting that in a future PR?

Reply:
Of the testing I've done, the os.Lstat calls are under 3% of the total runtime (according to the profile).

I'd be open to making the change in a subsequent PR, I wanted to keep this one focused to just the parallelism